tests/tests_geomstats/test_positive_lower_triangular_matrices.py

```diff
@@ -2,287 +2,323 @@
-        result = self.space.belongs(mats)
-        expected = True
-        self.assertAllClose(result, expected)
-
-        mats = gs.array([[1.0, -1.0], [-1.0, 3.0]])
-        result = self.space.belongs(mats)
-        expected = False
-        self.assertAllClose(result, expected)
-
-        mats = gs.array([[-1.0, 0.0], [-1.0, 3.0]])
-        result = self.space.belongs(mats)
-        expected = False
-        self.assertAllClose(result, expected)
-
-        mats = gs.eye(3)
-        result = self.space.belongs(mats)
-        expected = False
-        self.assertAllClose(result, expected)
-
-    def test_belongs_vectorization(self):
-        """Test of belongs method."""
-        mats_2dim = gs.array(
-            [
-                [[1.0, 0], [0, 1.0]],
-                [[1.0, 2.0], [2.0, 1.0]],
-                [[-1.0, 0.0], [1.0, 1.0]],
-                [[0.0, 0.0], [1.0, 1.0]],
+    space = PositiveLowerTriangularMatrices
+
+    class TestDataPositiveLowerTriangularMatrices(TestData):
+        def belongs_data(self):
+            smoke_data = [
+                dict(n=2, mat=[[1.0, 0.0], [-1.0, 3.0]], expected=True),
+                dict(n=2, mat=[[1.0, -1.0], [-1.0, 3.0]], expected=False),
+                dict(n=2, mat=[[-1.0, 0.0], [-1.0, 3.0]], expected=False),
+                dict(n=3, mat=[[1.0, 0], [0, 1.0]], expected=False),
+                dict(
+                    n=2,
+                    mat=[
+                        [[1.0, 0], [0, 1.0]],
+                        [[1.0, 2.0], [2.0, 1.0]],
+                        [[-1.0, 0.0], [1.0, 1.0]],
+                        [[0.0, 0.0], [1.0, 1.0]],
+                    ],
+                    expected=[True, False, False, False],
+                ),
+                dict(
+                    n=3,
+                    mat=[
+                        [[1.0, 0.0, 1.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+                        [[0.0, 0.0, 0.0], [2.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+                        [[1.0, 0.0, 0.0], [2.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+                        [[-1.0, 0.0, 0.0], [2.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+                    ],
+                    expected=[False, False, True, False],
+                ),
             ]
+            return self.generate_tests(smoke_data)
+
+        def random_point_and_belongs_data(self):
+            smoke_data = [
+                dict(n=1, n_points=1),
+                dict(n=2, n_points=2),
+                dict(n=10, n_points=100),
+                dict(n=100, n_points=10),
+            ]
+            return self.generate_tests(smoke_data)
+
+        def gram_data(self):
+            smoke_data = [
+                dict(
+                    n=2,
+                    point=[[1.0, 0.0], [2.0, 1.0]],
+                    expected=[[1.0, 2.0], [2.0, 5.0]],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def differential_gram_data(self):
```

Comment:
Same: no testing for batch data ("vectorization") for differential_gram?

Reply:
there is test_differential_gram_belongs

Reply:
But I will add batch deterministic data

Reply:
Ditto